documentation/modules/upgrading/proc-upgrade-brokers-newer-kafka.adoc

```diff
@@ -52,7 +52,9 @@ If the `log.message.format.version` and `inter.broker.protocol.version` are unse
 +
 NOTE: The value of `log.message.format.version` and `inter.broker.protocol.version` must be strings to prevent them from being interpreted as floating point numbers.
 
-. Change the `Kafka.spec.kafka.version` to specify the new version (leaving the `log.message.format.version` and `inter.broker.protocol.version` at the _current_ versions).
+. Change the `Kafka.spec.kafka.version` to specify the new Kafka version.
```

Comment:
Unless it's already explained elsewhere, I think it would be helpful to explain here that this change is about ensuring that all the brokers in the cluster are using the new binary, and during this change they need to continue using the old IBP so that the brokers can continue communicating.

Reply:
Yes, I think that's useful to know at this stage. I added a note:

> NOTE: Changing the `kafka.version` ensures that all brokers in the cluster will be upgraded to start using the new broker binaries. During this process, some brokers are using the old binaries while others have already upgraded to the new ones. Leaving the `inter.broker.protocol.version` unchanged ensures that the brokers can continue to communicate with each other throughout the upgrade.